libtelemetry/src/main/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -556,6 +554,8 @@ private void stopLocation() {
   private boolean unbindServiceConnection() {
     if (TelemetryUtils.isServiceRunning(TelemetryService.class)) {
       applicationContext.unbindService(serviceConnection);
```

Comment:
is there a reason we're using both `startService/stopService` and `bindService/unbindService`? Btw, calling `unbindService` on the service that was started with `startService` may not actually stop the service until you call `service.stopSelf()`.

Reply:
are we confident that this is not the root cause of the issue we're trying to solve? can you help me here to be a little less skeptical about the fix by adding unit test/instrumentation test?